### PR TITLE
Separate `limit` and `offset` operators from `select`.

### PR DESCRIPTION
**Pull request by sellout**
#done SD-1001, #done SD-1002

Also, fix `TransformSelect` to work on nested selects.

### CODE REVIEW

**Comment by sellout** [comment]
Assigned to @jedesah, but bring in @drostron on the review.

File: core/src/main/scala/quasar/std/set.scala [hunk 1]
```diff
@@ -24,7 +24,7 @@ import scalaz._, NonEmptyList.nel, Validation.{success, failure}
 import scalaz.syntax.applicative._
 
 trait SetLib extends Library {
-  val Take = Sifting("TAKE", "Takes the first N elements from a set", Type.Top :: Type.Int :: Nil,
+  val Take = Sifting("(LIMIT)", "Takes the first N elements from a set", Type.Top :: Type.Int :: Nil,
```
Comment (jedesah) [inline core/src/main/scala/quasar/std/set.scala]: Why the parenthesis around LIMIT?

Comment (jedesah) [inline core/src/main/scala/quasar/std/set.scala]: Can we add documentation explaining what `Sifting` represents.

Comment (sellout) [inline core/src/main/scala/quasar/std/set.scala]: The name of Funcs is currently an unfortunate conflation of its name and the name of the corresponding SQL function/operator. Basically we look up Funcs from the name of the SQL operation and the name has to match. It's something that needs to be fixed. And operators are displayed with parens.

Comment (sellout) [inline core/src/main/scala/quasar/std/set.scala]: I'll add docs for Sifting, but it's the same operations as ShapePreservingBuilder – they take a set and operate on the set itself, leaving individual elements unchanged – eg, sort, filter, take.

**Pull request merged**